### DISCUSSION TITLE
Add unique ID to `form.Field`

### DISCUSSION THREAD

**Discussion by Smeky** [TanStack/form#1007] [Ideas]
Hello 👋

Would it be possible to add some `uid` to fields? Main issue here is rendering Array fields. Currently the only way to set child key is to use the index from `field.state.value.map(_, i)`, as per docs.

This causes issues, mainly when removing items from said array, as the other fields in the array don't get correctly updated. 

https://react.dev/learn/rendering-lists#rules-of-keys

Hope I didn't miss something. In the meantime, could someone hint me at how to deal with this with some workaround? It's rather a minor thing but causes our forms to not be ready for production. 

Thanks & cheers!

**Comment by CanRau** [comment]
I think exposing a unique & SSR stable IDs per field `field.id` would also be nice for `<labels>`, without I have to wrap the label around the field, make a separate Field component or manage multiple `useId`.

This would be a nice QoL addition, if there's no overhead/downsides 🤔
